docs: add error codes auth (#22793)
supabase/supabase: apps/docs/docs/ref/shared/auth-error-codes.mdx

Supabase Auth can throw or return various errors when using the API. All errors originating from the `supabase.auth` namespace of the JavaScript client library will be wrapped by the `AuthError` class.

Error objects are split in a few classes:

- `AuthApiError` -- errors which originate from the Supabase Auth API.
  - Use `isAuthApiError` instead of `instanceof` checks to see if an error you caught is of this type.
- `CustomAuthError` -- errors which generally originate from state in the client library.
  - Use the `name` property on the error to identify the class of error received.

Errors originating from the server API classed as `AuthApiError` always have a `code` property that can be used to identify the error returned by the server. The `status` property is also present, encoding the HTTP status code received in the response.

In general the HTTP status codes you will likely receive are:

> - [403 Forbidden](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403) is sent out in rare situations where a certain Auth feature is not available for the user, and you as the developer are not checking a precondition whether that API is available for the user.

Reviewer: Consider adding examples for when a status code would surface

| `saml_assertion_no_email`               | SAML assertion (user information) was received after sign in, but no email address was found in it which is required. Check the provider's attribute mapping and/or configuration.                                                                                                                            |
| `saml_assertion_no_user_id`             | SAML assertion (user information) was received after sign in, but a user ID (called NameID) was not found in it which is required. Check the SAML identity provider's configuration.                                                                                                                          |
| `saml_entity_id_mismatch`               | (Admin API.) Updating the SAML metadata for a SAML identity provider is not possible, as the entity ID in the update does not match the entity ID in the database. This is equivalent to creating a new identity provider, and you should do that instead.                                                    |

> | `saml_idp_already_exists`               | (Admin API.) Adding a SAML identity provider that is already added.                                                                                                                                                                                                                                           |

Reviewer: Phone number already exists in the system.
```suggestion
| `saml_idp_already_exists`               | (Admin API.) SAML identity provider already exists in the system.                                                                                                                                                             |
```

| `saml_relay_state_expired`              | SAML relay state is an object that tracks the progress of a `supabase.auth.signInWithSSO()` request. The SAML identity provider should respond after a fixed amount of time, after which this error is shown. Ask the user to sign in again.                                                                  |
| `saml_relay_state_not_found`            | SAML relay states are progressively cleaned up after they expire, which can cause this error. Ask the user to sign in again.                                                                                                                                                                                  |
| `session_not_found`                     | Session to which the API request relates no longer exists. This can occur if the user has signed out, or the session entry in the database was deleted in some other way.                                                                                                                                     |

> | `signup_disabled`                       | Sign ups (new account creation) is disabled on the server.                                                                                                                                                                                                                                                    |

Reviewer: ```suggestion
| `signup_disabled`                       | Sign ups (new account creation) are disabled on the server.                                                                                                                                                                                                                                                    |
```

| `validation_failed`                     | Provided parameters are not in the expected format.                                                                                                                                                                                                                                                           |
| `weak_password`                         | User is signing up or changing their password without meeting the password strength criteria. Use the `AuthWeakPasswordError` class to access more information about what they need to do to make the assword pass.                                                                                           |

### Tips for better error handling

- Do not use string matching on error messages! Always use the `name` and `code` properties of error objects to identify the situation.

> - Although HTTP status codes generally don't change, they can suddenly change due to bugs, so avoid relying on them unless absolutely necessary.

Reviewer: ```suggestion
- HTTP codes may change due to bugs so avoid relying on them unless absolutely necessary
```
